manual/ec2-sru-20.2.txt

```diff
@@ -0,0 +1,222 @@
+    attachment_id_2=$(attach_network_interface $instance_id $network_interface_id_2 2)
+    sleep 60
+
+    ssh $sshopts $ec2_inst -- sudo cloud-init init
+    ssh $sshopts $ec2_inst -- ! grep Trace "/var/log/cloud-init*"
+    ssh $sshopts $ec2_inst -- sudo cloud-init clean --logs --reboot || true
+    sleep 60
+
+    if [ "$series" = "xenial" ]; then
+        ssh $sshopts $ec2_inst -- cat /etc/network/interfaces.d/50-cloud-init.cfg
+        net_file=/etc/network/interfaces.d/50-cloud-init.cfg
+    else
+        net_file=/etc/netplan/50-cloud-init.yaml
+    fi
+
+    perform_eni_route_verification $ec2_inst $net_file
+
+    teardown_network_interface $attachment_id_1 $network_interface_id_1
+    teardown_network_interface $attachment_id_2 $network_interface_id_2
+}
+
+
+for series in $SRU_SERIES; do
+    echo "### BEGIN $series"
+    EC2_INST=`launch-ec2 --series $series -u sethostname.yaml -t t2.small | awk '/Found/{print $5}'`
+    [ -n "$EC2_INST" ] || { echo "Failed to launch ec2 instance!"; exit 1; }
+    sleep 60
+    ssh $sshopts $EC2_INST -- cloud-init status --wait --long
+    ssh $sshopts $EC2_INST -- cat /run/cloud-init/result.json
+    ssh $sshopts $EC2_INST -- ! grep Trace "/var/log/cloud-init.*"
+    echo "Expect IMDSv2 token headers are being used on all metadata gets. Expect 0 line count"
+    ssh $sshopts $EC2_INST --  '[[ 0 = `grep headers /var/log/cloud-init.log  | grep -c -v X-aws` ]] || exit 1'
+    ssh $sshopts $EC2_INST -- sudo systemd-analyze
+    ssh $sshopts $EC2_INST -- sudo systemd-analyze blame
+    ssh $sshopts $EC2_INST -- sudo cloud-init analyze show
+    ssh $sshopts $EC2_INST -- ! sudo grep Trace "/var/log/cloud-init*"
+    scp $sshopts $PROPOSED_SCRIPT $EC2_INST:
+    ssh $sshopts $EC2_INST sudo bash $PROPOSED_SCRIPT 2>&1 | egrep 'cloud-init'
+    ssh $sshopts $EC2_INST sudo hostname something-else
+    ssh $sshopts $EC2_INST -- sudo cloud-init init
+    ssh $sshopts $EC2_INST -- ! grep Trace "/var/log/cloud-init*"
+    ssh $sshopts $EC2_INST -- sudo cloud-init clean --logs --reboot || true
+    sleep 60
+    date --utc +%s.%N
+    ssh $sshopts $EC2_INST -- cloud-init status --wait --long; date --utc +%s.%N
+    ssh $sshopts $EC2_INST -- hostname
+    ssh $sshopts $EC2_INST -- ! grep Trace "/var/log/cloud-init*"
+    ssh $sshopts $EC2_INST -- sudo systemd-analyze
+    ssh $sshopts $EC2_INST -- sudo cloud-init analyze show
+    if [ "$series" = "xenial" ]; then
+       echo "--- Expect warning on xenial for jinja because deb DOES NOT have jinja dep."
+    else
+       echo "--- Expect success on bionic for jinja because deb DOES have jinja dep."
+    fi
+    ssh $sshopts $EC2_INST "cloud-init query --format 'cloud-region: {{cloud_name}}-{{region}}'"
+    
+    echo 'Get cloud-id'
+    ssh $sshopts $EC2_INST cloud-id
+
+    echo "Per LP:1876323 expect do see correct output for cloud-init analyze"
+    ssh $sshopts $EC2_INST -- cloud-init analyze show | grep "Total Time"
```

Comment:
This bug is not verifiable on Ubuntu releases (which you are launching via your `launch-ec2 --series $series` on line 174. 

Amazon Linux is a different Centos/Fedora rpm-based distribution so it's different  than when we call launch-ec2 and spin up an Ubuntu image.

Also Ryan Harper already validated this particular bug with an individual test that exercised what we expect amazon linux 2 log configuration to look like (though he used lxcs to test) https://github.com/cloud-init/ubuntu-sru/blob/9129f45eed386d9074ab8ad5f404adf1271dfeba/bugs/lp-1876323.txt

Reply:
Makes total sense, I completely missed that this was not affecting ubuntu ec2 instances. I will drop this then